### ISSUE TITLE
Post Processing Failing on first checked file

### ISSUE THREAD

**Issue by hubbcaps** [evilhero/mylar#2179]
Mylar is continuing to snatch and download issues fine enough, however when it attempts to begin post processing an issue, it gets got on the first one, tracebacks and no longer attempts to post process anything

```
2019-01-29 17:28:54 | ERROR | Uncaught exception: Traceback (most recent call last):File "/app/mylar/mylar/logger.py", line 337, in new_runold_run(*args, **kwargs)File "/usr/lib/python2.7/threading.py", line 754, in runself.__target(*self.__args, **self.__kwargs)File "/app/mylar/mylar/PostProcessor.py", line 943, in Process"Type": av['Type'],IndexError: No item with that key
-- | -- | --
2019-01-29 17:28:54 | DEBUG | [POST-PROCESSING][SUCCESSFUL MATCH: Marvel Super Hero Contest of Champions-3147] Match verified for Marvel Super Hero Contest of Champions 002 (1982) (Digital) (Shadowcat-Empire).cbz
2019-01-29 17:28:54 | DEBUG | [POST-PROCESSING][DEFINITIVE-NAME MATCH] Definitive name match exactly to : Marvel Super Hero Contest of Champions [3147]
2019-01-29 17:28:54 | DEBUG | [POST-PROCESSING][SUB: Marvel.Super.Hero.Contest.of.Champions.002.1982.Digital.Shadowcat-Empire][CLOCATION: /downloads]
2019-01-29 17:28:54 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY][Lone Volume MATCH] Volume label of None indicates only volume for this series on your watchlist.
2019-01-29 17:28:54 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] Now checking against Marvel Super Hero Contest of Champions-3147
```

Running the linuxserver/mylar docker container, however I've modified it to be running on the development branch so I could get the fix you put in for me around the non numerical issue numbers.

If I move the file it fails on out of the directory, it does the same with the next:

```
2019-01-29 17:35:11 | ERROR | Uncaught exception: Traceback (most recent call last):File "/app/mylar/mylar/logger.py", line 337, in new_runold_run(*args, **kwargs)File "/usr/lib/python2.7/threading.py", line 754, in runself.__target(*self.__args, **self.__kwargs)File "/app/mylar/mylar/PostProcessor.py", line 943, in Process"Type": av['Type'],IndexError: No item with that key
-- | -- | --
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][SUCCESSFUL MATCH: New Mutants-26327] Match verified for New Mutants v3 020 (2011) (digital-hd) (Kleio-Empire).cbz
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][DEFINITIVE-NAME MATCH] Definitive name match exactly to : New Mutants [26327]
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][SUB: New.Mutants.v3.020.2011.digital-hd.Kleio-Empire.1][CLOCATION: /downloads]
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY][SeriesYear-Volume MATCH] Volume label of series Year of v3 matched to volume label of v3
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] 2011 comparing to 2011 : rechecking by month-check versus year.
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] IssueYear (issyr) is 2011
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] 2010-12-29 is before the issue year of 2011 that was discovered in the filename
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] Now checking against New Mutants-26327
2019-01-29 17:35:11 | DEBUG | temploc: 020
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][NON-MATCH: The New Mutants-3234] Incorrect series - not populating..continuing post-processing
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] 1984-10-01 is before the issue year 2011 that was discovered in the filename
2019-01-29 17:35:11 | DEBUG | [POST-PROCESSING][ISSUE-VERIFY] Now checking against The New Mutants-3234
```

I just use complete download handling, no scripts in Sab or anything.

Let me know if you need more info.

**Comment by hubbcaps** [comment]
I tried to get around this issue by using the Scan Comic Library/Import Results Management feature.

It worked for some of the comics, ended up renaming them and moving them into the proper folders in my actualy library from my downloads directory, however it didn't mark them as downloaded and they still appear as snatched now so that's gonna be fun working out which ones actually got imported vs which didn't lol

**Comment by BRTPOB** [comment]
You know, hubcaps, it would be easier to get help if you decided to hop back onto IRC.

**Comment by hubbcaps** [comment]
I know man, I'll get my shit reset up this weekend >.<

**Comment by evilhero** [comment]
Sorry about the delay - got stuck doing something else in regards to Mylar and only had a chance now to look at this. Was a simple fix, so going forward you won't get those errors any more when post-processing. 

Not quite sure how you fix the subsequent problem you encountered tho...

**Comment by hubbcaps** [comment]
No worries dude, I'm just happy you're here to fix things at all, and this is faster than most people would fix things. Really appreciate you.

I'll figure out the other problem, once everything else is matched up whatever's left I can manually sort through easy enough =]

**Comment by hubbcaps** [comment]
So it gets through the traceback now, but as soon as it hits that issue again, I start getting this repeated over and over in the logs with nothing else

```
[Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'The Contest', 'IssueID': u'22234', 'Publisher': u'Marvel', 'ReadingOrder': 1, 'IssueNumber': u'1', 'IssueDate': u'1982-06-01', 'Status': u'Wanted', 'ComicName': u'Marvel Super Hero Contest of Champions', 'DynamicComicName': u'marvelsuperherocontestofchampions', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'44423', 'IssueArcID': u'44423_2815', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'3', 'SeriesYear': u'1982', 'ComicVersion': 'v1982', 'ComicID': u'3147', 'Type': None, 'IsArc': True, 'LatestDate': u'1982-06-01'}]
```

**Comment by evilhero** [comment]
It should still be completing the process - that error line I'll have to suppress, but it's more like just an informational error line, it doesn't stop the post-processing and should continue. 

Granted if you have quite a few matches (depending on the series and arc's you have on your watchlist) it might spam the logs quite abit.

**Comment by hubbcaps** [comment]
I have debug but it doesn't seem to actually be printing the actual exception, and it's actually not post processing anything. The manual post process scans everything under the directory, and then when the post process begins, it hits this and spams the logs with this one issue over and over, no extra issues:

```
30-Jan-2019 18:28:32 - INFO    :: mylar.Process.534 : Thread-12 : Now checking: New Mutants [26327]
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.633 : Thread-12 : temploc: 020
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.717 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] Now checking against The New Mutants-3234
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.735 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] 1984-10-01 is before the issue year 2011 that was discovered in the filename
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.849 : Thread-12 : [POST-PROCESSING][NON-MATCH: The New Mutants-3234] Incorrect series - not populating..continuing post-processing
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.633 : Thread-12 : temploc: 020
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.717 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] Now checking against New Mutants-26327
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.731 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] 2010-12-29 is before the issue year of 2011 that was discovered in the filename
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.740 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] IssueYear (issyr) is 2011
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.745 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY] 2011 comparing to 2011 : rechecking by month-check versus year.
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.775 : Thread-12 : [POST-PROCESSING][ISSUE-VERIFY][SeriesYear-Volume MATCH] Volume label of series Year of v3 matched to volume label of v3
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.807 : Thread-12 : [POST-PROCESSING][SUB: /New.Mutants.v3.020.2011.digital-hd.Kleio-Empire.1][CLOCATION: /downloads]
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.860 : Thread-12 : [POST-PROCESSING][DEFINITIVE-NAME MATCH] Definitive name match exactly to : New Mutants [26327]
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.868 : Thread-12 : [POST-PROCESSING][SUCCESSFUL MATCH: New Mutants-26327] Match verified for New Mutants v3 020 (2011) (digital-hd) (Kleio-Empire).cbz
30-Jan-2019 18:28:32 - DEBUG   :: mylar.Process.965 : Thread-12 : [POST-PROCESSING] Now Checking if 1 issue(s) may also reside in one of the storyarc's that I am watching.
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
30-Jan-2019 18:28:32 - WARNING :: mylar.Process.989 : Thread-12 : [Exception: 'Type'] Unable to properly match up/retrieve issue number (or volume) for this [CS: {'StoryArc': u'Dark Reign', 'IssueID': u'202486', 'Publisher': u'Marvel', 'ReadingOrder': 460, 'IssueNumber': u'11', 'IssueDate': u'2010-05-01', 'Status': u'Skipped', 'ComicName': u'New Mutants', 'DynamicComicName': u'newmutants', 'IssueYear': None, 'ComicPublisher': u'Marvel', 'StoryArcID': u'55707', 'IssueArcID': u'55707_4073', 'Location': None}] [WATCHMATCH: {'Publisher': u'Marvel', 'Total': u'515', 'SeriesYear': u'2009', 'ComicVersion': 'v2009', 'ComicID': u'26327', 'Type': None, 'IsArc': True, 'LatestDate': u'2010-05-01'}]
```

**Comment by evilhero** [comment]
Ok, I really shouldn't be coding stuff without a nice hot jug of coffee beside me. Was referencing the incorrect variable, which was resulting in the type error - as it didn't exist in the section I was referring to. All fixed now in dev 5016fe1 !

**Comment by hubbcaps** [comment]
haha I understand that feeling man, no coffee and I tend to accidentally bring prod down. Thanks for the quick fix, it appears to be spinning through the PP now pretty smoothly as far as I can tell. Gonna let it run through it while I'm at work and see what I'm at tonight.
